content/docs/tasks/security/mtls-migration/index.md

```diff
@@ -0,0 +1,150 @@
+
+    ```command
+    $ kubectl get policies.authentication.istio.io --all-namespaces
+    No resources found.
+    $ kubectl get destionationrule --all-namespaces
+    No resources found.
+    ```
+
+## Configure the server to accept both mTLS and plain text traffic
+
+In Authentication Policy, we have a `PERMISSIVE` mode which makes the server accept both mutual TLS and plain text traffic.
+We need to configure the server to this mode.
+
+```bash
+cat <<EOF | istioctl create -n foo -f -
+apiVersion: "authentication.istio.io/v1alpha1"
+kind: "Policy"
+metadata:
+  name: "example-httpbin-permissive"
+  namespace: foo
+spec:
+  targets:
+  - name: httpbin
+  peers:
+  - mtls:
+      mode: PERMISSIVE
+EOF
+```
+
+Now send traffic to `httpbin.foo` again to ensure all requests can still succeed.
+
+```command
+$ for from in "foo" "bar" "legacy"; do kubectl exec $(kubectl get pod -l app=sleep -n ${from} -o jsonpath={.items..metadata.name}) -c sleep -n ${from} -- curl http://httpbin.foo:8000/ip -s -o /dev/null -w "sleep.${from} to httpbin.foo: %{http_code}\n"; done
+200
+200
+200
+```
+
+## Configure clients to send mutual TLS traffic
+
+Configure Istio services to send mutual TLS traffic by setting `DestinationRule`.
+
+```bash
+cat <<EOF | istioctl create -n foo -f -
+apiVersion: "networking.istio.io/v1alpha3"
+kind: "DestinationRule"
+metadata:
+  name: "example-httpbin-istio-client-mtls"
+spec:
+  host: httpbin.foo.svc.cluster.local
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+EOF
+```
+
+`sleep.foo` and `sleep.bar` should start sending mutual TLS traffic to `httpbin.foo`.
+
+Now we confirm all requests to `httpbin.foo` still succeed.
+
+```command
+$ kubectl exec $(kubectl get pod -l app=sleep -n bar -o jsonpath={.items..metadata.name}) -c sleep -n bar -- curl http://httpbin.foo:8000/ip -s -o /dev/null -w "%{http_code}\n"
+200
+```
+
+You can also specify a subset of the clients' request to use `ISTIO_MUTUAL` mutual TLS in
+[DestinationRule](https://istio.io/docs/reference/config/istio.networking.v1alpha3/#DestinationRule).
+After verifying it works by checking [Grafana to monitor](https://istio.io/docs/tasks/telemetry/using-istio-dashboard/),
+then increase the rollout scope and finally apply to all Istio client services.
+
+## Lock down to mutual TLS (optional)
+
+After migrating all clients to Istio services, injecting Envoy sidecar, we can lock down the `httpbin.foo` to only accept mutual TLS traffic.
+
+```bash
+cat <<EOF | istioctl create -n foo -f -
+apiVersion: "authentication.istio.io/v1alpha1"
+kind: "Policy"
+metadata:
+  name: "example-httpbin-permissive"
+  namespace: foo
+spec:
+  targets:
+  - name: httpbin
+  peers:
+  - mtls:
+      mode: STRICT
+EOF
+```
+
+Now you should see the request from `sleep.legacy` fails.
+
+```command
+$ kubectl exec $(kubectl get pod -l app=sleep -n bar -o jsonpath={.items..metadata.name}) -c sleep -n bar -- curl http://httpbin.foo:8000/ip -s -o /dev/null -w "%{http_code}\n"
+200
```

Comment:
still 200?

Reply:
Changed to 503. (need to confirm actually).